system/include/emscripten/wasmfs.h

```diff
@@ -92,6 +98,10 @@ backend_t wasmfs_create_root_dir(void);
 // backend is present there).
 void wasmfs_before_preload(void);
 
+#ifdef __cplusplus
+} // namespace wasmfs
+#endif
```

Comment:
Lets put this namespace only around the `backend_t` definition to avoid the complexity of having the functions be namespaced or not depending on the language. Now that I think about it, this conditional namespacing seems similarly complicated for the typedef as well...

Reply:
I think it makes the most sense to use forward declarations for now and add a TODO based on https://github.com/emscripten-core/emscripten/pull/19825#discussion_r1275392176.